### PR TITLE
Upgrade dependencies and fix testsuite

### PR DESCRIPTION
**Pull request by stes**
Fixes several issues with the test suite:

- Fix an issue in `test_criterions`: Most likely due to some change to the torch version (?), running the `test_infonce_gradients` test twice causes non-deterministic behavior. On the first run, non-existent gradients are `None` as expected, but on the second run, they are set to a zero torch array. I added a hotfix and accept both `None` and a zero array as possible values now, but this needs to deeper digging to fix properly.
- The `usage.rst` docs had a typo following the intro of discrete multisession training ( #135 ), which was fixed by introducing a separate variable name for the discrete multisession model there.
- Docs building failed. Removed pinning of `scikit-learn` and added a pin to `numpy<2` instead. In a future PR, we should look into upgrading some of the docs packages like `sphinx`, the theme, etc.
- Fix the `tools/build_docker.sh` script. I upgraded the docker container to a more recent python/cuda version and fixed the build process.

---

Fix https://github.com/AdaptiveMotorControlLab/CEBRA/pull/150
Fix https://github.com/AdaptiveMotorControlLab/CEBRA/pull/171

### CODE REVIEW

**Review by MMathisLab — APPROVED** [review]
✨🪄🦓
